opendevin/core/logger.py

```diff
@@ -42,7 +44,12 @@
 
 class ColoredFormatter(logging.Formatter):
     def format(self, record):
-        msg_type = record.__dict__.get('msg_type', None)
+        msg_type = record.__dict__.get('msg_type')
+        event_source = record.__dict__.get('event_source')
```

Comment:
Will the record have 'event_source'? If the record is based on an event, then the property name is `source`, isn't it?

I'm a bit curious in what scenario this worked for you. 😅

Reply:
I added it.

Reply:
Ah, right, you added it in session.py in this PR.

For visibility, for other reviewers: from private convo, I understand @SmartManoj aimed to fix or prepare to fix in this PR point 2 of this: https://github.com/OpenDevin/OpenDevin/pull/2493#pullrequestreview-2128934866

That makes sense to me, and now we changed the name of this PR to make it more clear it's a bit more broad than fixing the duplicate logging.